src/porting_android.h

```diff
@@ -76,4 +76,9 @@ int getInputDialogState();
  */
 std::string getInputDialogValue();
 
+#ifndef SERVER
+float getDisplayDensity();
```

Comment:
i think those functions should be part of RenderingEngine class

Reply:
RenderingEngine already has wrapper functions for these https://github.com/minetest/minetest/pull/6796/files#diff-b6a7a4bb97008d7e6dfe76e614fbfb02R574